### PR TITLE
Display ids

### PR DESCRIPTION
**Pull request by ahmedqarmout2**
This includes the following fixes:
Update  backend to use and handle datebase userIds
Admin page > visibility is hidden by default
Use display names for user friendly usage
Students page > make the entire question box clickable
Update datagen to use database Ids
Organise the scripts into their own directory
and other minor fixes over all

### CODE REVIEW

File: views/leaderboard-table.pug [hunk 1]
```diff
@@ -20,20 +20,20 @@ table.bordered
             td ...
             td ...
       else
-        if s.points != 0 && s.rank <= 3
-          if s.rank == 1
+        if s.points !== 0 && s.rank <= 3
```
Comment (ahmedqarmout2) [inline views/leaderboard-table.pug:23]: will fix the formating issues here

File: public/js/admin.js [hunk 2]
```diff
@@ -446,13 +448,13 @@ var deactivateUser = function(id) {
             success: function(data) {
                 displayAccountsTable();
                 const msg = ' has been&nbsp;<u><b>deactivated</b></u>&nbsp;';
-                warningSnackbar(id + ' account' + msg);
+                warningSnackbar(username + ' account' + msg);
```
Comment (esteeroramy) [inline public/js/admin.js]: why dont we put the message in directly?

File: public/js/admin.js [hunk 3]
```diff
@@ -605,7 +607,8 @@ var updateVisibility = function(qid) {
         function (isConfirm) {
             // User confirms the visiblity change
             if (isConfirm) {
-                question['visible'] = $('#checked-' + qid).is(':checked');
+                var visibilityStatus = $('#checked-' + qid).is(':checked');
```
Comment (esteeroramy) [inline public/js/admin.js]: 2 things
* why do we need a var here, why cant we do 
``` javascript
question['visible'] = $('#checked-' + qid).is(':checked');
```
* if we do need this variable, please change to `const`

File: public/js/admin.js [hunk 4]
```diff
@@ -614,22 +617,21 @@ var updateVisibility = function(qid) {
...
+                            warningSnackbar('Question ' + questionNumber + msg);
                         }
                     },
                     error: function(data) {
                         if (data['status'] === 401) {
                             window.location.href = '/';
                         } else {
-                            displayQuestionTable();
                             // Toast notification
+                            $('#checked-' + qid).prop('checked', !visibilityStatus);
```
Comment (esteeroramy) [inline public/js/admin.js]: we can get rid of `visibilityStatus` and just use `question['visible']` like you have above

File: scripts/dataGenerator.js [hunk 5]
```diff
@@ -117,31 +120,33 @@ var createStudents = function () {
...
+        email: common.formatString('{0}.{1}@mail.utoronto.ca', [firstName, lastName])
     };
 
     users.addStudent(acc, function (err, account) {
-        if (err) {
-            logger.error(common.formatString('Could not create account {0}. Please try again.', [accid]));
-        } else if (err === 'exists') {
+        if (err === 'exists') {
             logger.log(common.formatString('Account with username {0} exists.', [accid]));
+        }else if (err) {
```
Comment (esteeroramy) [inline scripts/dataGenerator.js]: space before else

File: server/common.js [hunk 6]
```diff
@@ -168,15 +158,20 @@ exports.isEmptyObject = isEmptyObject;
...
 
 // return the current date with format
 var getDateByFormat = function(format) {
     return date().format(format);
 }
-
-exports.getDate = getDate;
 exports.getDateByFormat = getDateByFormat;
 
+// get UUID
```
Comment (esteeroramy) [inline server/common.js]: lets start having proper documentation
``` javascript
/**
* get a random UUID
*
* @return {string}
*/

File: server/db.js [hunk 7]
```diff
@@ -264,6 +257,10 @@ var updateUserById = function(userId, info, callback){
         update.$set.email = info.email;
     }
 
+    if (info.points) {
+        update.$set.points = parseInt(info.points);
```
Comment (esteeroramy) [inline server/db.js]: lets make sure that this doesnt break if `info.points` is not an int

**Review by esteeroramy — COMMENTED** [review]
minor changes, please take a look, good work

File: public/js/admin.js (hunk 2 above)
Comment (ahmedqarmout2) [reply public/js/admin.js]: will do

File: public/js/admin.js (hunk 4 above)
Comment (ahmedqarmout2) [reply public/js/admin.js]: good point

**Comment by esteeroramy** [comment]
@ahmedqarmout2 please add the `fixes` keyword to this PR in order to close the tickets

**Review by Mohamed-A-Mohamed — COMMENTED** [review]
Looks good

**Review by Mohamed-A-Mohamed — APPROVED** [review]

**Review by Mohammed-JS-Khan — APPROVED** [review]

File: server/db.js [hunk 8]
```diff
@@ -241,27 +241,27 @@ var updateUserById = function(userId, info, callback){
...
         update.$set.lname = info.lname;
     }
 
-    if (info.email) {
+    if ('email' in info) {
         update.$set.email = info.email;
     }
 
-    if (info.points) {
+    if ('points' in info && parseInt(info.points)) {
```
Comment (esteeroramy) [inline server/db.js]: wouldnt this still break? should we make it `try {} catch() {}`

**Comment by ahmedqarmout2** [comment]
fixes #132 
fixes #67

**Review by esteeroramy — APPROVED** [review]

**Pull request merged**